fms/models/llama.py

```diff
@@ -396,10 +396,49 @@ def _rename_weights_to_fms(orig_sd):
     return new_sd
 
 
+def _hf_sd_to_fms_sd(hf_sd: OrderedDict):
+    replacements = [
+        (r"^lm_head.weight", "shared.head.weight"),
+        (r"^model.embed_tokens.weight", "shared.emb.weight"),
+        (r"^model.norm", "dec_norm"),
+        (r"^model.layers", "layers"),
+        (r"self_attn\.k_proj", "attn.key"),
+        (r"self_attn\.v_proj", "attn.value"),
+        (r"self_attn\.q_proj", "attn.query"),
+        (r"self_attn\.o_proj", "attn.dense"),
+        (r"mlp\.gate_proj", "ff_sub_layer.wg"),
+        (r"mlp\.up_proj", "ff_sub_layer.w1"),
+        (r"mlp\.down_proj", "ff_sub_layer.w2"),
+        (r"input_layernorm", "ln"),
+        (r"post_attention_layernorm", "ff_ln"),
+    ]
+    new_sd = {}
+    rotary_inv_freq_size = hf_sd["model.layers.0.self_attn.rotary_emb.inv_freq"].size(0)
```

Comment:
I think I'd prefer to just get emb_dim from each tensor's shape as-needed below, and use `emb_dim / 128`. That way if we try to use this to load certain submodules without loading the entire model, it'll still work. I think it'd also be a bit simpler to understand that llama's nheads == emb_dim / 128, vs what's in inv_freq.